src/commands/CountryCommand.ts

```diff
@@ -0,0 +1,391 @@
+
+export let formatNumber: (n: number) => string;
+
+class CountryCommand extends Command {
+	constructor() {
+		super("country");
+		initializeCountryData();
+	}
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		formatNumber = (n: number) => {
+			return n.toLocaleString(interaction.locale);
+		};
+
+		if (!countryData) {
+			interaction.reply({ content: "Still initiliazing the data, try again later...", ephemeral: true });
+			return;
+		}
+
+		switch (interaction.options.getSubcommand()) {
+			case "overview": {
+				const country: Country | undefined = getCountryByName(interaction.options.getString("country") ?? "Bhutan");
+				if (country == undefined) {
+					countryUndefinedReply(interaction);
+				} else {
+					interaction.reply({ embeds: [getOverviewEmbed(country)] });
+				}
+				break;
+			}
+			case "random": {
+				interaction.reply({ embeds: [getOverviewEmbed(countryData[Math.floor(Math.random() * countryData.length)])] });
+				break;
+			}
+			case "specific": {
+				const country: Country | undefined = getCountryByName(interaction.options.getString("country") ?? "Bhutan");
+				const info: string = interaction.options.getString("info") ?? "population";
+
+				if (country == undefined) {
+					countryUndefinedReply(interaction);
+				} else {
+					interaction.reply(specificRequestReplies[info](country));
+				}
+				break;
+			}
+			case "query": {
+				const sortCriteria: CountryKey = (interaction.options.getString("sort-criteria") ?? "none") as CountryKey;
+				const order = interaction.options.getString("order") ?? "ascending";
+				const scale = interaction.options.getInteger("scale") ?? 10;
+				const filterCriteria: CountryKey = (interaction.options.getString("filter-criteria") ?? "none") as CountryKey;
+				const relation = interaction.options.getString("relation") ?? "eq";
+				const filterValue = interaction.options.getString("filter-value") ?? "";
+				let includeData = interaction.options.getBoolean("include-data") ?? true;
+				let numbered = false;
+				let embedDataCriteria: CountryKey = sortCriteria;
+
+				// sorting
+				if ((sortCriteria as string) !== "none") {
+					numbered = true;
+					sortCountryDataBy(sortCriteria);
+					if (order === "descending") {
+						countryData.reverse();
+					}
+				} else {
+					countryData.sort(() => Math.random() - 0.5); // shuffle
+					includeData = false;
+				}
+
+				// filtering
+				let data: Country[] = countryData;
+				let val: number | string | boolean = filterValue;
+				if ((filterCriteria as string) !== "none") {
+					if (["true", "false"].includes(filterValue)) {
+						val = filterValue === "true";
+					} else if (!isNaN(+filterValue)) {
+						val = +filterValue;
+					}
+
+					if (
+						typeof val !== typeOfCountryProperty(filterCriteria) &&
+						!(typeof val === "string" && typeOfCountryProperty(filterCriteria) === "object")
+					) {
+						thatDoesntMakeSenseReply(interaction);
+						break;
+					}
```

Comment:
I assume this is for typings reasons? Could probably be fixed somehow better but no idea how to...

Reply:
No this is really part of the command it checks if the value entered for the filtering even matches the thing you're filtering by. 
For example you want to filter all the countries that use some currency but for a value you enter 3 
that doesn't make sense and this gets blocked here